src/Core__Array.res

```diff
@@ -7,18 +7,18 @@ external setUnsafe: (array<'a>, int, 'a) => unit = "%array_unsafe_set"
 @val
 external fromArrayLikeWithMap: (Js.Array2.array_like<'a>, 'a => 'b) => array<'b> = "Array.from"
 
-@send external fillAllInPlace: (array<'a>, 'a) => unit = "fill"
+@send external fillAll: (array<'a>, 'a) => unit = "fill"
 
-@send external fillInPlaceToEnd: (array<'a>, 'a, ~start: int) => unit = "fill"
+@send external fillToEnd: (array<'a>, 'a, ~start: int) => unit = "fill"
 
-@send external fillInPlace: (array<'a>, 'a, ~start: int, ~end: int) => unit = "fill"
+@send external fill: (array<'a>, 'a, ~start: int, ~end: int) => unit = "fill"
```

Comment:
Once the uncurry matures, these will all be subsumed into `fill`, without trailing `()`. The our one-off `toEnd` suffixes can all be (gently) deprecated too.